index.js

```diff
@@ -1,29 +1,48 @@
 const passport     = require('passport');
 const ldapStrategy = require('passport-ldapauth');
+const allowedAttributes = [
+  'dn', 'controls', 'objectClass', 'ou', 'o', 'ibmSerialNumber', 'employeeCountryCode', 
+  'isManager', 'managerCountryCode', 'co', 'uid', 'c', 'serialNumber', 'hrActive', 
+  'hrCompanyCode', 'hrCountryCode', 'hrFirstName', 'hrManagerIndicator', 'emailAddress', 
+  'mail', 'notesEmail', 'notesId', 'notesMailDomain', 'notesShortName', 'primaryNode', 'primaryUserId', 
+  'infotelephonenumber', 'personaltitle', 'jobresponsibilities', 'hrEmployeeType', 'employeeType', 
+  'preferredIdentity', 'iptelephonenumber', 'telephoneNumber', 'tieline', 'alternateTelephoneNumber', 
+  'mobile', 'timestampBpgui', 'hrLastName', 'coreDataIntegrity', 'directoryAlias', 'odmEntry', 'div', 
+  'notesMailFile', 'notesMailServer', 'buildingName', 'floor', 'ibmLoc', 'physicalDeliveryOfficeName', 
+  'workLoc', 'workLocation', 'workplaceIndicator', 'givenName', 'sn', 'timestampFeed', 'costCenter', 
+  'dept', 'divDept', 'hrOrganizationCode', 'callupName', 'cn', 'entryType', 'hrOrgUnitCode', 'manager', 
+  'managerSerialNumber', 'timestampHRMS', 'preferredLastName', 'timestampBpguiRI', 'passwordIsExpired', 
+  'passwordisreset', 'passwordIsResetByAdmin', 'passwordIsStruckOut', 'passwordModifyTimestamp']
 
 // Setup passport with IBM's LDAP config
 const config = { server: {
     url: 'ldaps://bluepages.ibm.com:636',
     searchBase: 'ou=bluepages,o=ibm.com',
-    searchFilter: '(mail={{username}})',
-    searchAttributes: ['cn', 'uid', 'mail']
+    searchFilter: '(mail={{username}})'
 }}
-passport.use(new ldapStrategy(config))
-passport.initialize()
 
-module.exports = function(req, res, next) {
+module.exports = {
+  initialize: function(attributes) {
```

Comment:
well about ur comments:

1. You're right, if you don't pass attributes it will deliver all attributes from LDAP Strategy API.
2. I think it is... because u aren't just setting attributes... since we are working with middlewere I think it must be initialized with the express and injected there in the middle somewhere (but I can be wrong, possibly). If you look the entire code you will see after setting the attributes (if exists) it will use the passport within the express (`passport.use(new ldapStrategy(config))`) and then initilize it (`passport.initialize()`) that's the reason I called this method as initilize.

But related with point 2, I'm opened to other names... I just want to explain why I called `initilize`